### PR TITLE
EZP-27615: If user does not have permission to view versions display message

### PR DESCRIPTION
**Pull request by maxbaldanza**
https://jira.ez.no/browse/EZP-27615

screen shot 2017-07-12 at 15 55 58

### CODE REVIEW

**Comment by bdunogier** [comment]
Should the buttons ("New draft based on select version", "Delete") be hidden when there are no versions displayed ?

**Comment by maxbaldanza** [comment]
Not according to the mock up on the ticket although I think this may be a little weird and have said the same thing on the ticket: https://jira.ez.no/browse/EZP-27615

**Comment by maxbaldanza** [comment]
@bdunogier buttons are now disabled as requested by @SylvainGuittard

**Comment by SylvainGuittard** [comment]
thanks @maxbaldanza

File: src/bundle/Controller/VersionController.php [hunk 1]
```diff
@@ -35,22 +35,19 @@ public function __construct(
...
-        $versions = $this->uiVersionService->loadVersions($contentInfo);
+        $versionParameterSupplier->supply($view);
 
-        $draftVersions = $versionFilter->filterDrafts($versions);
-        $draftActionsForm = $formFactory->createVersionsDraftActionForm($draftVersions);
-
-        $archivedVersions = $versionFilter->filterArchived($versions);
-        $archivedActionsForm = $formFactory->createVersionsArchivedActionForm($archivedVersions);
+        $draftActionsForm = $formFactory->createVersionsDraftActionForm(
+            $view->hasParameter('draftVersions') ? $view->getParameter('draftVersions') : []
```
Comment (bdunogier) [inline src/bundle/Controller/VersionController.php:39]: I really don't like that way of reusing what the supplier will hopefully set. There is no contract with the interface that certifies any of that.

**Review by bdunogier — CHANGES_REQUESTED** [review]
The way the supplier's data is used in the controller really annoys me. Please try to think of a better approach to this.

File: src/bundle/Controller/VersionController.php (hunk 1 above)
Comment (maxbaldanza) [reply src/bundle/Controller/VersionController.php:39]: 🤔 not sure there is a better approach without breaking the interface. Did you have something else in mind?
Our views assume that these values are set so isn't it safe for our forms to do the same?

**Review by bdunogier — APPROVED** [review]

**Pull request merged**